pkg/app/piped/livestatestore/cloudrun/store.go

```diff
@@ -44,54 +93,81 @@ func (s *store) run(ctx context.Context) error {
 		// https://cloud.google.com/run/quotas#api
 		v, next, err := s.client.List(ctx, ops)
 		if err != nil {
-			return fmt.Errorf("failed to list cloudrun services: %w", err)
+			return nil, err
 		}
-		svc = append(svc, v...)
+		svcs = append(svcs, v...)
 		if next == "" {
 			break
 		}
 		cursor = next
 	}
-
-	// Update apps to the latest.
-	s.setApps(svc)
-
-	return nil
+	return svcs, nil
 }
 
-func (s *store) setApps(svc []*provider.Service) {
-	apps := make(map[string]provider.ServiceManifest, len(svc))
-	for i := range svc {
-		sm, err := svc[i].ServiceManifest()
-		if err != nil {
-			s.logger.Error("failed to load cloudrun service into service manifest", zap.Error(err))
+func (s *store) listRevisionsFromServices(ctx context.Context, svcs []*provider.Service) map[string][]*provider.Revision {
+	revs := make(map[string][]*provider.Revision, len(svcs))
+	for i := range svcs {
+		id, ok := svcs[i].UID()
+		if !ok {
 			continue
 		}
-		appID := sm.Labels()[provider.LabelApplication]
-		apps[appID] = sm
+		names := svcs[i].ActiveRevisionNames()
+		if rs := s.getMultiRevisions(ctx, names); len(rs) > 0 {
+			revs[id] = rs
+		}
 	}
-	s.apps.Store(apps)
+	return revs
+}
+
+func (s *store) getMultiRevisions(ctx context.Context, names []string) []*provider.Revision {
+	ops := &provider.ListRevisionsOptions{
+		LabelSelector: provider.MakeRevisionNamesSelector(names),
+	}
+	v, _, err := s.client.ListRevisions(ctx, ops)
+	if err != nil {
+		s.logger.Error("failed to list cloudrun revisions", zap.Error(err))
+	}
+	return v
 }
 
-func (s *store) loadApps() map[string]provider.ServiceManifest {
+func (s *store) loadApps() map[string]*app {
 	apps := s.apps.Load()
 	if apps == nil {
 		return nil
 	}
-	return apps.(map[string]provider.ServiceManifest)
+	return apps.(map[string]*app)
 }
 
-func (s *store) GetServiceManifest(appID string) (provider.ServiceManifest, bool) {
+func (s *store) getServiceManifest(appID string) (provider.ServiceManifest, bool) {
 	apps := s.loadApps()
 	if apps == nil {
 		s.logger.Error("failed to load cloudrun apps")
 		return provider.ServiceManifest{}, false
 	}
 
-	sm, ok := apps[appID]
+	app, ok := apps[appID]
 	if !ok {
 		return provider.ServiceManifest{}, false
 	}
 
-	return sm, true
+	return app.service, true
+}
+
+func (s *store) getState(appID string) (State, bool) {
+	apps := s.loadApps()
+	if apps == nil {
+		s.logger.Error("failed to load cloudrun apps")
```

Comment:
Do we need this error log?

Reply:
TBH, It is not necessary because it print out logs with the upstream.
Thank you!